assets/check

```diff
@@ -35,23 +36,29 @@ fi
 if private_registry "${repository}" ; then
   registry="$(extract_registry "${repository}")"
   repository="$(extract_repository "${repository}")"
-  protocol=http
 else
   registry=registry.hub.docker.com
-  protocol=https
 fi
 
+if [ "$server_args" != "${server_args%--insecure-registry=*}" ]; then
+  curl_opts="${curl_opts} -k"
+fi
+
+protocol="$(find_protocol "${registry}")"
 uri_root="${protocol}://${registry}"
 
 tags=$(mktemp /tmp/resource-in-tags.XXXXXX)
 curl ${curl_opts} "${uri_root}/v1/repositories/${repository}/tags" > $tags
 
-shortened_image=$(jq -r "map(select(.name == $(echo $tag | jq -R .)))[].layer" < $tags)
-
-all_images=$(mktemp /tmp/resource-in-tags.XXXXXX)
-curl ${curl_opts} "${uri_root}/v1/repositories/${repository}/images" > $all_images
+shortened_image=$(jq -r "try map(select(.name == $(echo $tag | jq -R .)))[].layer" < $tags)
 
-latest_image=$(jq -r "map(select(.id | startswith($(echo $shortened_image | jq -R .))))[0].id // \"\"" < $all_images)
+if [ -n shortend_image ]; then
+  all_images=$(mktemp /tmp/resource-in-tags.XXXXXX)
+  curl ${curl_opts} "${uri_root}/v1/repositories/${repository}/images" > $all_images
+  latest_image=$(jq -r "map(select(.id | startswith($(echo $shortened_image | jq -R .))))[0].id // \"\"" < $all_images)
+else
+  latest_image=$(jq -r "$(echo $tag) | jq -R .")
```

Comment:
What are these changes for? Looks unrelated to the rest (also this line has broken syntax).